app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :likes
+  has_many :comments
+
+  def update_post_counter
+    user.update(posts_counter: Post.where(user_id: user).count)
+  end
```

Comment:
- [OPTIONAL] This is similar to the methods discussed above. You may consider refactoring this one too. 
```suggestion
  def update_post_counter
    author.increment!(:posts_counter)
  end
```

Reply:
Well solved